### PR TITLE
Delete unused deps

### PR DESCRIPTION
**Pull request by neil-da**
I gave it a run through `weeder` which found a bunch of little stuff, most of which we were relying on, but it pointed out a few cleanups.

### CODE REVIEW

File: daml-foundations/daml-ghc/ide/src/Development/IDE/State/Rules/Daml.hs [hunk 1]
```diff
@@ -186,7 +192,7 @@ generateDalfRule =
         unsimplifiedRawDalf <- use_ GenerateRawDalf file
         let rawDalf = LF.simplifyModule unsimplifiedRawDalf
         setPriority PriorityGenerateDalf
-        pure $ toIdeResult $ do
+        pure $ either (, Nothing) (([],) . Just) $ do
```
Comment (cocreature) [inline daml-foundations/daml-ghc/ide/src/Development/IDE/State/Rules/Daml.hs]: Not sure duplicating this logic is an improvement.

Comment (neil-da) [reply daml-foundations/daml-ghc/ide/src/Development/IDE/State/Rules/Daml.hs]: I'm not convinced this pattern should exist anywhere - it should really be fused with whatever is producing it. By making it internal we can do that more easily. I suspect there's a further cleanup to be done here, but was aiming for one step at a time.

Comment (cocreature) [reply daml-foundations/daml-ghc/ide/src/Development/IDE/State/Rules/Daml.hs]: Having operations that can only emit errors but not warnings doesn’t seem that unreasonable to me but I don’t feel particularly strongly about this.

Comment (neil-da) [reply daml-foundations/daml-ghc/ide/src/Development/IDE/State/Rules/Daml.hs]: OK, I think you've convinced me. I put that bit back. Not sure `toIdeResult` is the right name, but we can solve that in future.

Comment (cocreature) [reply daml-foundations/daml-ghc/ide/src/Development/IDE/State/Rules/Daml.hs]: I full agree that `toIdeResult` is a terrible name, I was actually trying to change it recently but failed to come up with something better.

**Pull request merged**